src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/routinghelper/RoutingHelperAction.java

```diff
@@ -0,0 +1,78 @@
+package org.openstreetmap.josm.plugins.pt_assistant.actions.routinghelper;
+
+import java.awt.event.ActionEvent;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.swing.JOptionPane;
+
+import com.drew.lang.annotations.NotNull;
+import com.drew.lang.annotations.Nullable;
+import org.openstreetmap.josm.data.osm.Relation;
+import org.openstreetmap.josm.data.osm.Way;
+import org.openstreetmap.josm.gui.MainApplication;
+import org.openstreetmap.josm.gui.MapFrame;
+import org.openstreetmap.josm.gui.dialogs.relation.actions.AbstractRelationEditorAction;
+import org.openstreetmap.josm.gui.dialogs.relation.actions.IRelationEditorActionAccess;
+import org.openstreetmap.josm.gui.dialogs.relation.actions.IRelationEditorUpdateOn;
+import org.openstreetmap.josm.plugins.pt_assistant.utils.BoundsUtils;
+import org.openstreetmap.josm.tools.I18n;
+import org.openstreetmap.josm.tools.ImageProvider;
+
+public class RoutingHelperAction extends AbstractRelationEditorAction {
+    private static final Set<ITransportMode> TRANSPORT_MODES = Collections.singleton(new BusTransportMode());
```

Comment:
There's now a set of transport modes (for now just bus, others could follow later). The transport mode handles everything specific to the transport mode, like determining if you are allowed to travel along a certain way.

The routing helper selects the appropriate mode for a relation and queries the transport mode when there is something specific to the transport mode.